aten/src/ATen/native/LinearAlgebraUtils.h

```diff
@@ -576,4 +576,24 @@ static inline bool linalg_solve_is_vector_rhs(const Tensor& input, const Tensor&
   return vector_case;
 }
 
+static inline bool is_blas_compatible_column_major_order(const Tensor& input) {
+  IntArrayRef input_strides = input.strides();
+  IntArrayRef input_sizes = input.sizes();
+  auto ndim = input.dim();
+  TORCH_INTERNAL_ASSERT_DEBUG_ONLY(ndim == 2);
```

Comment:
This might be something we'll need to revisit if https://github.com/pytorch/pytorch/issues/63583 isn't resolved. We can merge it for now, but beware.